src/mlpack/methods/cf/interpolation_policies/regression_interpolation.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file regression_interpolation.hpp
+ * @author Wenhao Huang
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_REGRESSION_INTERPOLATION_HPP
+#define MLPACK_METHODS_CF_REGRESSION_INTERPOLATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * 
+ */
+class RegressionInterpolation
+{
+ public:
+  /**
+   *
+   */
+  RegressionInterpolation() { }
+
+  /**
+   * 
+   */
+  RegressionInterpolation(const arma::sp_mat& cleanedData)
+  {
+    const size_t userNum = cleanedData.n_cols;
+    a.set_size(userNum, userNum);
```

Comment:
You can also write: ``a.set_size(arma::size(cleanedData))``, just wanted to point that out.